clippy_lints/src/loops/never_loop.rs

```diff
@@ -43,6 +50,14 @@ pub(super) fn check<'tcx>(
     }
 }
 
+fn is_loop_contains_break_continue(expr: &Expr<'_>) -> bool {
+    for_each_expr_without_closures(expr, |e| match e.kind {
+        ExprKind::Break(..) | ExprKind::Continue(..) => ControlFlow::Break(()),
+        _ => ControlFlow::Continue(()),
```

Comment:
You should ignore break or continue from inner loops as they won't prevent from applying the fix:

```suggestion
        ExprKind::Break(..) | ExprKind::Continue(..) => ControlFlow::Break(()),
        ExprKind::Loop(..) => ControlFlow::Continue(Descend::No),
        _ => ControlFlow::Continue(Descend::Yes),
```

and add a test case such as:

```rust
fn no_break_or_continue_loop_outer() {
    for i in [1, 2, 3].iter() {
        //~^ never_loop
        return;
        loop {
            if true {
                continue;
            }
        }
    }
}
```

Reply:
Also, if you apply this fix (and I think you should), it should be done only when the loop has no label, or an inner `break` or `continue` with this label would fail.

Reply:
Thanks! I've implemented it in such a way that it takes into account whether the loop is labeled or not when determining applicability. The existing tests are sufficient for this, so I did not add any new ones.